modules/core/arrow-extras-data/src/main/kotlin/arrow/data/fingertree/FingerTree.kt

```diff
@@ -0,0 +1,57 @@
+package arrow.data.fingertree
+
+import arrow.core.Option
+import arrow.data.fingertree.internal.FingerTreeInternal
+
+class FingerTree<T> internal constructor(internal: FingerTreeInternal<T> = FingerTreeInternal.Empty()) {
```

Comment:
Shouldn't this be `private var fingerTreeInternal: FingerTreeInternal<T>?`

Reply:
I implemented it that way to make the class `FingerTree` testable. 
Please let me know if you know a better way 😀